### PR TITLE
Fix `Jitter` transform with integer array input

### LINKED ISSUE #12404: Jitter example is broken
ref: https://docs.bokeh.org/en/dev-3.0/docs/examples/plotting/jitter.html

The "jitter" plot has no jitter applied.

cc @mattpap not sure what is going on, since another jitter example *is* working:

https://docs.bokeh.org/en/dev-3.0/docs/examples/plotting/categorical_scatter_jitter.html

### PR DESCRIPTION
**Pull request by mattpap**
This also exposes random number generator API and allows to configure a random number generator in `Jitter`.

fixes #12404

### CODE REVIEW

File: bokehjs/src/lib/core/util/math.ts [hunk 1]
```diff
@@ -75,26 +71,6 @@ export function to_radians_coeff(units: AngleUnits): number {
...
-    if (-4*r1*r1*Math.log(r1) >= r2*r2)
-      break
-  }
-  let rn = r2/r1
-
-  // Transform the standard normal to meet the characteristics that we want (mu, sigma)
-  rn = mu + sigma*rn
-
-  return rn
-}
```
Comment (mattpap) [inline bokehjs/src/lib/core/util/math.ts:96]: @ianthomas23, can you help me out with understanding if I can replace this function with `AbstractRandom.normals()` (see below)? In general, can you review changes related to random number generation.

**Comment by bryevdv** [comment]
Offhand, I don't really like the idea of adding this as a model. There's a tension now between `distribution` and `random` (which should really be called `random_generator`) but mostly it also just seems unnecessary. No one has ever once asked for this to be parameterizable, and the use-case for it seems vanishingly small and also out of scope. Rather than take on new API commitments without a clear motivation, I would at this point, rather just pick a method to use internally in BokehJS and document that choice.

File: bokehjs/src/lib/core/util/math.ts (hunk 1 above)
Comment (ianthomas23) [reply bokehjs/src/lib/core/util/math.ts:96]: Yes, you can replace this with `AbstractRandom.normals()`, the two functions are equivalent. This looks unnecessarily complicated whereas the Box-Muller method in `AbstractRandom.normals()` can be explained using simple geometry in polar coordinates and has a wikipedia page (https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform).

**Comment by ianthomas23** [comment]
IIRC some of the low-level changes here are clearing up mess of my making as the LaTeX normal distribution example used a new random number generator (the Box-Muller one) rather than the previous existing one. We really don't need both, which this fixes.

File: bokehjs/src/lib/core/util/math.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/src/lib/core/util/math.ts:96]: Good, this way we will have one source of random number generation in `core/util/random`.

**Comment by mattpap** [comment]
I renamed `random` to `random_generator` and made it an internal property (at least for now). I removed new models from public API.

**Comment by mattpap** [comment]
> No one has ever once asked for this to be parameterizable, and the use-case for it seems vanishingly small and also out of scope.

To me, having precise control over the behavior of a library like bokeh, is clearly in the scope of the project. Users may not have asked for this, as they didn't ask for majority of bokeh's features, yet we implemented them anyway. Quite often it's us creators that have to propose certain features, for users to realize that this is what they wanted all along. Also, transforms currently have limited usefulness, so I'm presuming that most users will choose to jitter and apply other transforms manually.

**Pull request merged**